docs/orchestration/faq/faq.md

```diff
@@ -0,0 +1,228 @@
+
+```python
+import pendulum
+from prefect.engine.signals import RETRY
+
+@task
+def my_sensor(**kwargs):
+    # check some state of the world
+    if condition_not_met:
+        raise RETRY("Condition not met, retrying in 5 seconds.", start_time=pendulum.now().add(seconds=5))
+```
+
+### How do I set a dynamic default value for a Parameter like a callable?
+
+If you do something like:
+
+```python
+from datetime import date
+
+with Flow("example") as flow:
+	start_date = Parameter("start_date", default=(date.today().strftime("%Y-%m-%d"))
+	do_something(start_date)
+```
+
+The default Parameter value is evaluated during registration time, so the start_date in the example above will be fixed to the registration date. In order to add dynamicism, using a task is needed to defer the execution of date.today() to runtime. For example:
+
+```python
+from datetime import date
+
+@task
+def get_date(day):
+    if day == None:
+        return date.today().strftime("%Y-%m-%d")
+    else:
+        return day
+
+with Flow("example") as flow:
+	start_date = Parameter("start_date", default=None)
+    new_date = get_date(start_date)
+	do_something(new_date)
+```
+
+### How can I set a custom flow run name?
+
+The flow run name can not be set in advance, but it can be changed using the [RenameFlowRun](api/latest/tasks/prefect.html#renameflowrun) task after the flow run has been created. You can use this task inside the Flow block, or through a flow-level state handler. When calling the task from a state handler, make sure to call the task’s  `.run()` method.
+
+Here is a flow-level state handler example:
+
+```python
+from prefect import Flow, task
+from prefect.tasks.prefect import RenameFlowRun
+
+def rename_handler(obj, new_state, old_state):
+    if new_state.is_running():
+        RenameFlowRun().run(flow_run_name="new_name")
+    return
+
+@task
+def first_task():
+    return 1
+
+with Flow("test-flow", state_handlers=[rename_handler]) as flow:
+    first_task()
+```
+
+### How do I add things to the prefect context so I can reuse them in other tasks?
+
+The Prefect context is not meant to be mutable so adding an item to it inside a task will not take effect in downstream tasks. Often, a much better approach is to use a Parameter to contain a value that will be used or manipulated by tasks.
+
+### How do I pull Cloud secrets instead of using Local Secrets?
+
+Flow runs triggered by an agent will pull Cloud secrets by default. If you want to pull Cloud secrets during testing with `flow.run()`, you can export the environment variable:
+
+```bash
+export PREFECT__CLOUD__USE_LOCAL_SECRETS=false
+```
+or you can use the `config.toml` setting
+```
+[cloud]
+use_local_secrets = false
+```
+
+### [REVIEW] Can I run a Docker agent in a container?
+
+In general, the Docker agent is supposed to run in a local process (rather than in a docker container), and this local process is a layer between Prefect backend and a Docker Daemon.
```

Comment:
```suggestion
In general, the Docker agent is supposed to run in a local process (rather than in a Docker container), and this local process is a layer between Prefect backend and a Docker daemon.
```